### PR TITLE
Add constant folding for FNegate

### PR DESCRIPTION
**Pull request by jaebaek**
Let `OpFNegate` against a constant be the negative of the operand.

Contributes to #709.

### CODE REVIEW

File: test/opt/fold_test.cpp [hunk 1]
```diff
@@ -799,7 +799,15 @@ ::testing::Values(
...
-        2, -std::numeric_limits<float>::infinity())
+        2, -std::numeric_limits<float>::infinity()),
+    // Test case 3: Fold -2.0
+    InstructionFoldingCase<float>(
+        Header() + "%main = OpFunction %void None %void_func\n" +
+            "%main_lab = OpLabel\n" +
+            "%2 = OpFNegate %float %float_2\n" +
+            "OpReturn\n" +
+            "OpFunctionEnd",
+        2, -2)
```
Comment (s-perron) [inline test/opt/fold_test.cpp]: I know we don't do this for many cases, but could you add a test for when the input is the result of and OpConstantNull.

Also add a test for a vector.

Comment (s-perron) [reply test/opt/fold_test.cpp]: Test double as well.

**Comment by dneto0** [comment]
Hi @jaebaek : Please rebased your change and add tests as requested

File: test/opt/fold_test.cpp (hunk 1 above)
Comment (jaebaek) [reply test/opt/fold_test.cpp]: Done

**Pull request merged**